dp/generate_climos.py

```diff
@@ -516,6 +531,87 @@ def update_metadata_and_time_var(input_file, t_start, t_end, operation, climo_fi
         if hasattr(input_file, 'tracking_id'):
             cf.climo_tracking_id = input_file.tracking_id
 
+        def compare_times(t1, t2):
```

Comment:
We might be able to simplify this considerably extracting the string-to-datetime function embedded here, and use it just to the convert (date) strings extracted from history, and compare the result in-place, e.g., `start < to_datetime(s)`. That would make the logic clearer and DRY up this code, which contains code for the date conversion repeated twice, once for each argument.

And we could also not convert the start/end time forward into a string, then back into a datetime for comparison; We could just pass the start/end time as a `datetime`, compare it without need for conversion, and convert it just once when it is written to the history as a string. It's almost always better to keep values in their natural computational representation (in this case `datetime`) internally, and only convert to strings for I/O.

Tell me if I am missing something here.